framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -747,4 +753,60 @@ public static String removeNewLines(String input) {
     return builder.toString();
   }
 
+  /* Get DrillTestCase for a single query file */
+  public static void singleTestCase(File singleTestFile,
+                                      String[] testGroups,
+                                      List<DrillTestCase> drillTestCases)
+      throws IOException {
+    List<File> jsonFiles = searchFiles(singleTestFile.getParentFile(), ".*.json");
+    if (jsonFiles.isEmpty()) {
+      String updir = "";
+      while (jsonFiles.isEmpty()) {
+        updir = updir + "/..";
+        File singleTestFileParent = new File(singleTestFile.getParent() + updir);
+        jsonFiles = searchFiles(singleTestFileParent, ".*.json");
+      }
+    }
+    for (File jsonFile : jsonFiles) {
+      TestCaseModeler modeler;
+      try {
+        modeler = getTestCaseModeler(jsonFile.getAbsolutePath());
+      } catch (JsonParseException e) {
+        LOG.warn("Caught exception parsing " + jsonFile + ". This test will not be executed.", e);
+        continue;
+      }
+
+      String queryFileExtension = modeler.matrices.get(0).inputFile;
+      Pattern pattern = Pattern.compile(queryFileExtension + "$");
+      Matcher matcher = pattern.matcher(singleTestFile.getName());
+      if (matcher.find()) {
+        List<String> categories = modeler.categories;
+        boolean foundTests = false;
+        for (String testGroup : testGroups) {
+          if (categories != null && !categories.contains(testGroup)) {
+            continue;
+          } else {
+            foundTests = true;
+            break;
+          }
+        }
+        if (!foundTests) {continue;}
+        String expectedFileExtension = modeler.matrices.get(0).expectedFile;
+        boolean skipSuite = false;
+        if (modeler.dependencies != null) {
+          for (String dependency : modeler.dependencies) {
+            if (TestDriver.cmdParam.excludeDependenciesAsList().contains(dependency)) {
+              skipSuite = true;
+            }
+          }
+        }
+        if (skipSuite) {continue;}
+        String expectedFileName = getExpectedFile(singleTestFile.getAbsolutePath(),
+                  queryFileExtension, expectedFileExtension);
+        drillTestCases.add(new DrillTestCase(modeler, singleTestFile.getAbsolutePath(), expectedFileName));
+        break;
+      }
+    }
+  }
+
```

Comment:
Can you test the following scenarios:
1. -s .../dir contains test definition file and sudirs containing .json
2. -s .../test_definition_file.json
3. -s .../test_query.q
4. -s .../dir without any test definition files in any of the dir and parent dirs

Reply:
Do we need to specify the relative path to the test, including the file extension?

Reply:
What does the "..." do?

1. how is .json different from test definition file?
2. are there any files besides test_definition_file.json in the directory?

Reply:
@Agirish.  Yes.  Previously we would say
   -s Functional/p1tests
Now we can say
   -s Functional/p1tests/is_null.q